File: python/valuecell/server/api/schemas/decision_effectiveness.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class DecisionEffectivenessBreakdownItem(BaseModel):
    label: str
    count: int
    effective_count: int
    partially_effective_count: int
    failed_count: int
    observing_count: int
    insufficient_count: int
    average_score: int


class DecisionEffectivenessRecentItem(BaseModel):
    ticker: str
    display_name: str
    action: str
    outcome_status: str
    summary: str
    review_date: str


class DecisionEffectivenessSummaryData(BaseModel):
    generated_at: datetime
    available: bool
    empty_message: str | None = None
    overall_summary: str
    overall_score: int
    review_count: int
    effective_count: int
    partially_effective_count: int
    failed_count: int
    observing_count: int
    insufficient_count: int
    action_breakdown: list[DecisionEffectivenessBreakdownItem] = Field(
        default_factory=list
    )
    role_breakdown: list[DecisionEffectivenessBreakdownItem] = Field(
        default_factory=list
    )
    theme_breakdown: list[DecisionEffectivenessBreakdownItem] = Field(
        default_factory=list
    )
    recent_successes: list[DecisionEffectivenessRecentItem] = Field(default_factory=list)
    recent_failures: list[DecisionEffectivenessRecentItem] = Field(default_factory=list)

File: frontend/src/types/decision-effectiveness.ts
export type DecisionEffectivenessBreakdownItem = {
  label: string;
  count: number;
  effective_count: number;
  partially_effective_count: number;
  failed_count: number;
  observing_count: number;
  insufficient_count: number;
  average_score: number;
};

export type DecisionEffectivenessRecentItem = {
  ticker: string;
  display_name: string;
  action: string;
  outcome_status: string;
  summary: string;
  review_date: string;
};

export type DecisionEffectivenessSummary = {
  generated_at: string;
  available: boolean;
  empty_message: string | null;
  overall_summary: string;
  overall_score: number;
  review_count: number;
  effective_count: number;
  partially_effective_count: number;
  failed_count: number;
  observing_count: number;
  insufficient_count: number;
  action_breakdown: DecisionEffectivenessBreakdownItem[];
  role_breakdown: DecisionEffectivenessBreakdownItem[];
  theme_breakdown: DecisionEffectivenessBreakdownItem[];
  recent_successes: DecisionEffectivenessRecentItem[];
  recent_failures: DecisionEffectivenessRecentItem[];
};
